### DISCUSSION TITLE
react-hook-form validate onChange

### DISCUSSION THREAD

**Discussion by darkpsinight** [react-hook-form/react-hook-form#9252] [Q&A] [answered]
Below is my form using `react-hook-form`, with two inputs (`username` and `email`). This form fire validation on `onSubmit` action (on button click).
How to make it fire on `input` `value` change only (on `onChange` event) instead?

```JSX
import React from 'react';
import { useForm } from 'react-hook-form';

export default function App() {
  const { register, handleSubmit, formState: { errors } } = useForm();
  const onSubmit = data => console.log(data);
  console.log(errors);
  
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <input type="text" placeholder="username" {...register("username", {required: true, maxLength: 80})} />
      <input type="undefined" placeholder="Email" {...register} />

      <input type="submit" />
    </form>
  );
}
```

**Comment by georgehwc** [comment]
just add onchange 
```
            <input 
              {...register("username", {
                required: true, maxLength: 80,
                onChange: (e) => { console.log(e.target.value) },
              })}
              type="text" placeholder="Type here"  />
```
not sure if this work, I just made another question, hope this help.
https://stackoverflow.com/questions/66936135/react-hook-form-how-to-can-i-use-onchange-on-react-hook-form-version-7-0

**Reply by Moshyfawn** [reply]
This is unrelated to the OP's question

**Comment by Moshyfawn** [comment, marked as answer]
```js
useForm({
  mode: "onChange"
})
```
does exactly that. Please, refer to [RHF docs](https://react-hook-form.com/api/useform) to learn more

**Reply by darkpsinight** [reply]
thank you

**Reply by philipaantony** [reply]
Thank you

**Comment by ghost** [comment]
Will this still work when passing in a resolver?
I'm wrapping my inputs in Controller
I've got mode: 'onChange' but only submitting causes validation
```
    const { control, handleSubmit, setValue } = useForm<SignUpFormDataType>({
        resolver: joiResolver(schema),
        mode: 'onChange',
    })
```
The only validation occurs on submit, even though I've specified onChange. 
Can post more code but have a feeling a resolver doesn't get called onChange.. will keep trying if thats not the case.

**Reply by gustavo-maurina** [reply]
Ever found a fix? I'm facing the same problem currently

**Reply by ghost** [reply]
Haven't looked at it in a while, but have bumped versions, will see if its still not working and get back to you.
